### PR TITLE
Add support for Vite development mode

### PR DESCRIPTION
**Pull request by remcohaszing**
This also makes the rollup plugin compatible with both Rollup and Vite on a type level by specifying the bare minimum.

Refs #2367

### CODE REVIEW

File: packages/rollup/test/index.js [hunk 1]
```diff
@@ -56,4 +58,41 @@ test('@mdx-js/rollup', async function (t) {
     await fs.rm(mdxUrl)
     await fs.rm(jsUrl)
   })
+
+  await t.test('vite production', async () => {
+    const result = /** @type {RollupOutput[]} */ (
```
Comment (wooorm) [inline packages/rollup/test/index.js:63]:
```suggestion
    const result = /** @type {Array<RollupOutput>} */ (
```

**Comment by wooorm** [comment]
No clue why Vercel failed? It should use `public` by default, which we make?

File: packages/rollup/lib/index.js [hunk 2]
```diff
@@ -19,6 +18,17 @@
...
+/**
+ * @typedef Plugin
+ *   A plugin that is compatible with both Rollup and Vite.
+ * @property {string} name
+ *   The name of the plugin
+ * @property {(config: unknown, env: { mode: string }) => undefined} config
+ *   A function used by Vite to set additional configuration options.
+ * @property {(value: string, path: string) => Promise<SourceDescription | undefined>} transform
+ *   A function to transform the source content.
+ */
```
[the remark this answers is not in the document]
Comment (remcohaszing) [reply packages/rollup/lib/index.js:30]: Ah yes, much better. This allows us to document the params.

File: packages/rollup/lib/index.js [hunk 3]
```diff
@@ -34,23 +44,35 @@ import {VFile} from 'vfile'
  */
```
[the remark this answers is not in the document]
Comment (remcohaszing) [reply packages/rollup/lib/index.js:67]: We could. I don’t find if very useful here, as this is covered by the test code.

**Comment by wooorm** [comment]
Can you fix the problem?

**Comment by remcohaszing** [comment]
Vercel seems to expect a `dist` directory, not `public`. It looks unrelated to these changes.

**Comment by wooorm** [comment]
yeah I posted that earlier: https://github.com/mdx-js/mdx/pull/2376#issuecomment-1772687723. No clue why it suddenly breaks

**Comment by wooorm** [comment]
They defaulted to `public`, which is why we use that folder.
Somehow apparently they now didn‘t default to that, but expected `dest`.
I added `public` explicitly on their site, and it works

**Comment by wooorm** [comment]
Thanks!

**Pull request merged**